src/linalg/Eig.cpp

```diff
@@ -92,4 +93,107 @@ namespace cytnx {
   }  // namespace linalg
 }  // namespace cytnx
 
+namespace cytnx {
+  namespace linalg {
+
+    // actual impls:
+    void _Eig_Dense_UT(std::vector<cytnx::UniTensor> &outCyT, const UniTensor &Tin,
+                       const bool &is_V, const bool &row_v) {
+      //[Note] outCyT must be empty!
+
+      // DenseUniTensor:
+      // cout << "entry Dense UT" << endl;
+
+      Tensor tmp;
+      if (Tin.is_contiguous())
+        tmp = Tin.get_block_();
+      else {
+        tmp = Tin.get_block();
+        tmp.contiguous_();
+      }
+
+      vector<cytnx_uint64> tmps = tmp.shape();
+      vector<cytnx_int64> oldshape(tmps.begin(), tmps.end());
+      tmps.clear();
+      vector<string> oldlabel = Tin.labels();
+
+      // collapse as Matrix:
+      cytnx_int64 rowdim = 1;
+      for (cytnx_uint64 i = 0; i < Tin.rowrank(); i++) rowdim *= tmp.shape()[i];
+      tmp.reshape_({rowdim, -1});
+
+      vector<Tensor> outT = cytnx::linalg::Eig(tmp, is_V, row_v);
+      if (Tin.is_contiguous()) tmp.reshape_(oldshape);
```

Comment:
Why reshape_ when contiguous?

Reply:
tmp refer to the Tin block so we need to reshape it back.